src/index.ts

```diff
@@ -235,14 +235,23 @@ function build_commit_string(commit_state: z.infer<typeof CommitState>, config:
      commit_string += ': '
   }
 
-  if (commit_state.ticket) {
-     commit_string += colorize ? color.magenta(commit_state.ticket) + ' ' : commit_state.ticket + ' '
-  }
+  if(config.check_ticket.add_to_title){
+
+    if(config.check_ticket.title_position === "start"){
+      commit_string += colorize ? color.magenta(commit_state.ticket) + ' ' : commit_state.ticket + ' '
+      commit_string += colorize ? color.reset(commit_state.title) : commit_state.title
+    }else {
+      commit_string += colorize ? color.reset(commit_state.title) + ' ' : commit_state.title + ' '
+      commit_string += colorize ? color.magenta(commit_state.ticket)  : commit_state.ticket 
+    }
+  }else{
 
-  if (commit_state.title) {
-    commit_string += colorize ? color.reset(commit_state.title) : commit_state.title;
+    if (commit_state.title) {
+      commit_string += colorize ? color.reset(commit_state.title) : commit_state.title;
+    }
   }
```

Comment:
I'm good with these changes. I would prefer this and avoid all the nesting

```javascript
  const position_start = config.check_ticket.title_position === "start"
  const position_end = config.check_ticket.title_position === "end"
  if(commit_state.ticket && config.check_ticket.add_to_title && position_start) {
    commit_string += colorize ? color.magenta(commit_state.ticket) + ' ' : commit_state.ticket + ' '
  }

  if (commit_state.title) {
    commit_string += colorize ? color.reset(commit_state.title) : commit_state.title;
  }

  if(commit_state.ticket && config.check_ticket.add_to_title && position_end) {
    commit_string += colorize ? color.magenta(commit_state.ticket) : commit_state.ticket
  }
```

Also I think you need to rebase, this PR is including your changes from yesterday.

Reply:
Also also, lol, working right now so I threw that together super quick, if you make that change above I would double check it 😅

Reply:
Nice job, good catch on the position_end if needing that space at the beginning :)